modules/lib/src/main/resources/assets/js/app/ui2/list/ContentIcon.tsx

```diff
@@ -24,61 +24,136 @@ import {
     FileMusic,
     Archive
 } from 'lucide-react';
-import {JSX} from 'react';
+import {useState, type JSX} from 'react';
+
+const ImgWithFallback = ({
+                             src,
+                             alt = '',
+                             fallback = <FileIcon/>,
+                             imgClassName,
+                         }: {
+    src?: string | null;
+    alt?: string;
+    fallback?: JSX.Element;
+    imgClassName?: string;
+}): JSX.Element => {
+    const [errored, setErrored] = useState(false);
+    if (!src || errored) {
+        return fallback;
+    }
+    return <img className={imgClassName} alt={alt} src={src} onError={() => setErrored(true)}/>;
+};
 
 interface Props {
     contentType: ContentTypeName;
+    imageUrl?: string | null;
 }
 
-export const ContentIcon = ({contentType}: Props): JSX.Element => {
+const fallbackIconFor = (contentType: ContentTypeName): JSX.Element => {
+    const key = String(contentType);
+    switch (key) {
+    case String(ContentTypeName.IMAGE):
+        return <FileImage/>;
+    case String(ContentTypeName.MEDIA_IMAGE):
+        return <ImageIcon/>;
+    case String(ContentTypeName.MEDIA_VECTOR):
+        return <SplinePointer/>;
+    default:
+        return <FileIcon/>;
+    }
+};
```

Comment:
It can stay here, but I don’t see any reason why it should look like a regular function instead of a component:

```tsx
const FallbackContentIcon = ({type}: {type: ContentTypeName}): JSX.Element => {
```